detectron2/engine/hooks.py

```diff
@@ -199,6 +201,81 @@ def after_step(self):
         self.step(self.trainer.iter)
 
 
+class BestCheckpointer(HookBase):
+    """
+    Checkpoints best weights based off given metric.
+
+    This hook should be used in conjunction to and executed after `EvalHook`.
+    """
+
+    def __init__(
+        self,
+        eval_period: int,
+        checkpointer: Checkpointer,
+        val_metric: str = "bbox/AP50",
+        file_prefix: str = "model_best",
+        append_iter: bool = False,
+    ) -> None:
+        """
+        Args:
+            eval_period (int): the period `EvalHook` is set to run.
+            checkpointer: the checkpointer object used to save checkpoints.
+            val_metric (str): chosen validation metric to track for best checkpointing
+            file_prefix (str): the prefix of checkpoint's filename, defaults to "model_best"
+            append_iter (bool): if True, checkpoint file name will have iteration num appended, else write to the same checkpoint file.
```

Comment:
There is no harm to save iter (it's saved in all other checkpoints anyway). Let's use True and remove this option.
However I don't think we should add iter to the saved file name - it will then save __many__ best models instead of one.